src/ucp/core/ucp_ep.c

```diff
@@ -2812,6 +2812,18 @@ int ucp_ep_do_keepalive(ucp_ep_h ep)
     return 0;
 }
 
+static void ucp_ep_req_purge_send(ucp_request_t *req, ucs_status_t status)
+{
+    if ((status == UCS_OK) ||
+        (ucp_ep_config(req->send.ep)->key.err_mode ==
+         UCP_ERR_HANDLING_MODE_NONE) ||
+        !(req->flags & UCP_REQUEST_FLAG_RKEY_INUSE)) {
+        ucp_request_complete_and_dereg_send(req, status);
+    } else {
+        ucp_request_dt_invalidate(req, status);
+    }
```

Comment:
```suggestion
    if ((status != UCS_OK) &&
        (ucp_ep_config(req->send.ep)->key.err_mode ==
         UCP_ERR_HANDLING_MODE_PEER) &&
        (req->flags & UCP_REQUEST_FLAG_RKEY_INUSE)) {
        ucp_request_dt_invalidate(req, status);
    }

    ucp_request_complete_and_dereg_send(req, status);
```

Reply:
will not work: request should not be completed prior to invalidation is complete.
and request_dt_invalidate is dereferencing region

Reply:
yeah, I missed `return;` after invalidate

Reply:
ok, done